File: scripts_analysis/00_import.jl
## ------------------------------------------------------------------------
##
## Script name: 00_import.jl
## Purpose: Import data and source scripts for analysis
## Author: Yanwen Wang
## Date Created: 2024-12-08
## Email: yanwenwang@u.nus.edu
##
## ------------------------------------------------------------------------
##
## Notes:
##
## ------------------------------------------------------------------------

# Load the required packages
using Arrow
using CategoricalArrays
using DataFrames
using DataFramesMeta
using Distributions
using FreqTables
using LinearAlgebra
using ProportionalFitting
using Random
using Statistics
using StatsBase

# Load functions
include("functions.jl")

# Load data
census = DataFrame(Arrow.Table("Data_clean/census.arrow"))
sample_women = DataFrame(Arrow.Table("Samples/sample_women.arrow"))
sample_men = DataFrame(Arrow.Table("Samples/sample_men.arrow"))
sample = DataFrame(Arrow.Table("Samples/sample.arrow"))

# Source scripts
# @time include("01_sample.jl")
@time include("02_trends.jl")
@time include("03_composition.jl")
@time include("04_gradient.jl")
@time include("06_decomp_prepare.jl")
@time include("07_decomp.jl")

File: scripts_visualize/auxiliary/00_import.r
## ------------------------------------------------------------------------
##
## Script name: 00_import.r
## Purpose: Import libraries and source scripts for visualization
## Author: Yanwen Wang
## Date Created: 2024-12-10
## Email: yanwenwang@u.nus.edu
##
## ------------------------------------------------------------------------
##
## Notes:
##
## ------------------------------------------------------------------------

# Load the required packages
library(arrow)
library(tidyverse)
library(patchwork)

theme_set(theme_bw())

# Source scripts
source("scripts_visualize/auxiliary/01_trends.r")
source("scripts_visualize/auxiliary/02_composition.r")
source("scripts_visualize/auxiliary/03_gradient.r")
source("scripts_visualize/auxiliary/04_odds_ratio.r")
source("scripts_visualize/auxiliary/05_decomp.r")
